tips/tip-0001/tip-0001.md

```diff
@@ -0,0 +1,67 @@
+If zero-knowledge is an objective, then $a(X) = a^*(X) + k \cdot T_{\mathsf{fd}}(X)$ and $b(X) = b^*(X) - k \cdot T_{\mathsf{rp}}(X)$, where $a^*(X)$ and $b^*(X)$ are the minimal-degree Bezout coefficients as returned by the extended Euclidean algorithm, and where $k \in \mathbb{F}$ is a uniformly random field element called the *Bezout randomizer*. For the sake of the soundness analysis, set the degree bound on $a(X)$ and $b(X)$ to $T$, the height of the table.
+
+When the verifier supplies $\alpha$, the prover responds with (among other things) the terminal values $T_{\mathsf{rp}} = T_{\mathsf{rp}}(\alpha)$ and $T_{\mathsf{fd}} = T_{\mathsf{fd}}(\alpha)$. The prover also supplies the Bezout coefficients $A = a(\alpha)$ and $B = b(\alpha)$. The verifier verifies the correct calculation of the terminals using the AIR, and the correct calculation of the Bezout coefficients using the DEEP technique: the prover adds $q_{a}(X) = \frac{a(X) - A}{X - \alpha}$ and $q_{b}(X) = \frac{b(X) - B}{X - \alpha}$ to the nonlinear combination, and the verifier verifies that it was added. The verifier additionally verifies that $A \cdot T_{\mathsf{rp}} + B \cdot T_{\mathsf{fd}} = 1$.
+
+**Correctness.** The prover is incapable of proving that the gcd equals 1 when $T_{\mathsf{rp}}(X)$ and $T_{\mathsf{df}}(X)$ share a factor. Since $g$ and $h$ are scalars they do not change the gcd. Therefore, the gcd can only be non-one for dishonest provers. As a result, the protocol has perfect completeness. $\square$
+
+**Soundness.** If the table has at least one non-contiguous region, then $T_{\mathsf{rp}}(X)$ and $T_{\mathsf{fd}}(X)$ share at least one factor. As a result, no Bezout coefficients $a(X)$ and $b(X)$ can exist such that $a(X)\cdot T_{\mathsf{rp}}(X) + b(X) \cdot T_{\mathsf{fd}}(X) = 1$. The verifier therefore probes an inequality polynomials of degree at most $2T$. According to the Schwartz-Zippel lemma the false positive probability is at most $2T / |\mathbb{F}|$. $\square$
+
+**Zero-Knowledge.** The prover sends four polynomial evaluations, two of which are terminals. We treat each value in turn.
+ 1. $T_{\mathsf{rp}}(\alpha)$: For any $T_{\mathsf{rp}}(\alpha)$, for any list $\lbrace \mathsf{ramp}_i \rbrace_i$ of accumulated $\mathsf{ramp}$ values, and for any challenge $\alpha$, there is an initial $g$ that is consistent with the given view. Indeed, the mapping from $g \mapsto T_{\mathsf{rp}}(\alpha)$ is affine and the coefficient is nonzero as long as $\alpha \not \in \lbrace\mathsf{ramp}_i\rbrace_i$.
+ 2. $T_ {\mathsf{fd}}(\alpha)$: For any pair $(T_ {\mathsf{rp}}(\alpha), T_ {\mathsf{fd}}(\alpha))$, for any list $\lbrace \mathsf{ramp}_ i \rbrace_ i$ of accumulated $\mathsf{ramp}$ values, and for any challenge $\alpha$, there is an initial $g$ that is consistent with the given view. Indeed, the mapping from $h \mapsto T_ {\mathsf{fd}}(\alpha)$ is affine and the coefficient is nonzero as long as $\alpha \not \in \lbrace\mathsf{ramp}_ i\rbrace_ i$.
```

Comment:
Is the initial in question really $g$, not $h$?

Reply:
The one in bullet point 2 should be $h$.